airbyte-integrations/connectors/source-snowflake/unit_tests/integration/test_discover.py

```diff
@@ -0,0 +1,226 @@
+from parameterized import parameterized
+
+from integration.config import ConfigBuilder
+from integration.request_builder import SnowflakeRequestBuilder
+from integration.response_builder import JsonPath
+from source_snowflake.source import SourceSnowflake
+
+
+
+_WAREHOUSE = "WAREHOUSE"
+_DATABASE = "DATABASE"
+_SCHEMA = "INTEGRATION_TEST"
+_TABLE = "TEST_TABLE"
+_ROLE = "ROLE"
+_HOST = "host.com"
+_REQUESTID = "123"
+_HANDLE = "123-DUMMY-HANDLE-123"
+
+
+def _config() -> ConfigBuilder:
+    return ConfigBuilder(jwt_token="123", host=_HOST, schema=_SCHEMA, database=_DATABASE, role=_ROLE, warehouse=_WAREHOUSE)
+
+
+def table_request() -> SnowflakeRequestBuilder:
+    return SnowflakeRequestBuilder.statement_endpoint(host=_HOST, schema=_SCHEMA, database=_DATABASE, role=_ROLE, warehouse=_WAREHOUSE)
+
+
+def _catalog(sync_mode: SyncMode) -> ConfiguredAirbyteCatalog:
+    return CatalogBuilder().build()
+
+
+def _source(catalog: ConfiguredAirbyteCatalog, config: Dict[str, Any], state: Optional[List[AirbyteStateMessage]]) -> SourceSnowflake:
+    return SourceSnowflake()
+
+
+def snowflake_response(file_name, datafield: Path = FieldPath("data")) -> SnowflakeResponseBuilder:
+    return create_response_builder(
+        find_template(file_name, __file__),
+        datafield,
+        pagination_strategy=SnowflakePaginationStrategy()
+    )
+
+
+def a_snowflake_response(file_name, datafield: Path = FieldPath("data"), id_path=None, cursor_path=None) -> RecordBuilder:
+    return create_record_builder(
+        find_template(file_name, __file__),
+        datafield,
+        record_id_path=id_path,
+        record_cursor_path=cursor_path
+    )
+
+
+def _given_table_catalog(http_mocker: HttpMocker) -> None:
+    http_mocker.post(
+        table_request().with_show_catalog().with_requestID(_REQUESTID).build(),
+        snowflake_response("check_connection").with_record(a_snowflake_response("check_connection")).build()
+    )
+
+
+def _given_table_with_primary_keys(http_mocker: HttpMocker, table) -> None:
+    http_mocker.post(
+        table_request().with_table(table).with_show_primary_keys().with_requestID(_REQUESTID).build(),
+        snowflake_response("primary_keys").with_record(a_snowflake_response("primary_keys")).build()
+    )
+
+def _given_read_schema(http_mocker: HttpMocker, table) -> None:
+    http_mocker.post(
+        table_request().with_table(table).with_get_schema().with_requestID(_REQUESTID).build(),
+        snowflake_response("response_get_table", JsonPath("$")).with_record(
+            a_snowflake_response("response_get_table", JsonPath("$"))).build()
+    )
+
+def _given_get_timezone(http_mocker: HttpMocker) -> None:
+    http_mocker.post(
+        table_request().with_requestID(_REQUESTID).with_timezone().build(),
+        snowflake_response("timezone", JsonPath("$.'data'")).build())
+
+
+def _discover(
+        config_builder: ConfigBuilder,
+        sync_mode: SyncMode,
+        state: Optional[List[AirbyteStateMessage]] = None,
```

Comment:
I would assume that there should be no state here as the discover does not take a state as an input